File: app/src/main/java/com/ianarbuckle/dublinbushelper/transports/schedules/ScheduleActivity.java
package com.ianarbuckle.dublinbushelper.transports.schedules;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.FragmentManager;
import android.view.MenuItem;

import com.ianarbuckle.dublinbushelper.BaseActivity;
import com.ianarbuckle.dublinbushelper.BaseFragment;
import com.ianarbuckle.dublinbushelper.R;
import com.ianarbuckle.dublinbushelper.utils.Constants;
import com.ianarbuckle.dublinbushelper.utils.UiUtils;

/**
 * Created by Ian Arbuckle on 02/03/2017.
 *
 */

public class ScheduleActivity extends BaseActivity {

  public static Intent newIntent(Context context) {
    return new Intent(context, ScheduleActivity.class);
  }

  @Override
  protected void onCreate(@Nullable Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    initFragment();
    initToolbar();
  }

  @Override
  protected void initLayout() {
    setContentView(R.layout.activity_container);
  }

  @Override
  protected void initToolbar() {
    super.initToolbar();
    String displayName = getIntent().getStringExtra(Constants.DISPLAYNAME_KEY);
    assert toolbar != null;
    UiUtils.customiseToolbar(toolbar);
    UiUtils.backStyleActionBar(toolbar);
    toolbar.setTitle(displayName);
    assert getSupportActionBar() != null;
    getSupportActionBar().setHomeButtonEnabled(true);
    getSupportActionBar().setHomeAsUpIndicator(UiUtils.backStyleActionBar(toolbar));

  }

  private void initFragment() {
    FragmentManager fragmentManager = getSupportFragmentManager();

    if(fragmentManager.findFragmentByTag(Constants.SCHEDULE_FRAGMENT) != null) {
      return;
    }

    BaseFragment.switchFragment(fragmentManager, ScheduleFragment.newInstance(), Constants.SCHEDULE_FRAGMENT, false);
  }

  @Override
  public boolean onOptionsItemSelected(MenuItem item) {
    switch(item.getItemId()) {
      case android.R.id.home:
        finish();
    }
    return true;
  }
}

File: app/src/main/java/com/ianarbuckle/dublinbushelper/transports/schedules/ScheduleActivity.kt
package com.ianarbuckle.dublinbushelper.transports.schedules

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.MenuItem
import com.ianarbuckle.dublinbushelper.BaseActivity

import com.ianarbuckle.dublinbushelper.BaseFragment
import com.ianarbuckle.dublinbushelper.R
import com.ianarbuckle.dublinbushelper.utils.Constants

/**
 * Created by Ian Arbuckle on 02/03/2017.

 */

class ScheduleActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        initFragment()
        initToolbar(true)
    }

    override fun initLayout() {
        setContentView(R.layout.activity_container)
    }

    override fun initToolbar(isHomeAsUpEnabled: Boolean) {
        super.initToolbar(isHomeAsUpEnabled)
        val displayName = intent.getStringExtra(Constants.DISPLAYNAME_KEY)
        toolbar?.title = displayName
    }

    private fun initFragment() {
        val fragmentManager = supportFragmentManager
        BaseFragment.switchFragment(fragmentManager, ScheduleFragment.newInstance(), Constants.SCHEDULE_FRAGMENT, false)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> finish()
        }
        return true
    }

    companion object {

        fun newIntent(context: Context): Intent {
            return Intent(context, ScheduleActivity::class.java)
        }
    }
}
